### ISSUE TITLE
panic: runtime error: invalid memory address or nil pointer dereference

### ISSUE THREAD

**Issue by fthoms** [getgauge/gauge#319]
Just installed gauge 0.4.0 in windows 10 Home 64bit. Tried both the 32 bit and 64 bit installer, but with the same result. Ran the following commands after install:

gauge --init csharp
gauge specs\example.spec

got the following result:
-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=
gauge specs\example.spec
Building Project: C:\Users\fthomsen\tmp\gt\gt.sln
Success
panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xc0000005 code=0x0 addr=0x0 pc=0x4d26cd]

goroutine 1 [running]:
github.com/getgauge/gauge/validation.(*specValidator).validateStep(0x132efec0, 0x1328a4d0, 0x132d0de0)
        /tmp/src/github.com/getgauge/gauge/validation/validate.go:239 +0x38d
github.com/getgauge/gauge/validation.(*specValidator).Step(0x132efec0, 0x1328a4d0)
        /tmp/src/github.com/getgauge/gauge/validation/validate.go:214 +0xd0
github.com/getgauge/gauge/validation.(*specValidator).ContextStep(0x132efec0, 0x1328a4d0)
        /tmp/src/github.com/getgauge/gauge/validation/validate.go:253 +0x29
github.com/getgauge/gauge/gauge.(*Specification).Traverse(0x131b4d00, 0x33220ac0, 0x132efec0)
        /tmp/src/github.com/getgauge/gauge/gauge/specification.go:202 +0x23a
github.com/getgauge/gauge/validation.(*specValidator).validate(0x132efec0, 0x0, 0x0, 0x0)
        /tmp/src/github.com/getgauge/gauge/validation/validate.go:201 +0x5d
github.com/getgauge/gauge/validation.(*validator).validate(0x130c1e8c, 0x0)
        /tmp/src/github.com/getgauge/gauge/validation/validate.go:189 +0x16f
github.com/getgauge/gauge/validation.ValidateSpecs(0x13068128, 0x1, 0x1, 0x1317bc20, 0x25, 0x331243b0)
        /tmp/src/github.com/getgauge/gauge/validation/validate.go:101 +0x127
github.com/getgauge/gauge/execution.ExecuteSpecs(0x13068128, 0x1, 0x1, 0x0)
        /tmp/src/github.com/getgauge/gauge/execution/execute.go:82 +0xb5
main.main()
        /Users/stechops/Library/Application Support/Go Agent/pipelines/Gauge-Package/gauge-compile/gauge.go:137 +0x851

goroutine 33 [select]:
net/http.(*persistConn).roundTrip(0x1318e240, 0x131fa450, 0x0, 0x0, 0x0)
        /usr/local/Cellar/go/1.5.1/libexec/src/net/http/transport.go:1164 +0x932
net/http.(*Transport).RoundTrip(0x130947e0, 0x131eb110, 0xb2, 0x0, 0x0)
        /usr/local/Cellar/go/1.5.1/libexec/src/net/http/transport.go:235 +0x42b
net/http.send(0x131eb110, 0x33124288, 0x130947e0, 0x131eb180, 0x0, 0x0)
        /usr/local/Cellar/go/1.5.1/libexec/src/net/http/client.go:220 +0x40a
net/http.(*Client).send(0xbdff20, 0x131eb110, 0x63, 0x0, 0x0)
        /usr/local/Cellar/go/1.5.1/libexec/src/net/http/client.go:143 +0xf5
net/http.(*Client).doFollowingRedirects(0xbdff20, 0x131eb110, 0xa2f0a4, 0x0, 0x0, 0x0)
        /usr/local/Cellar/go/1.5.1/libexec/src/net/http/client.go:380 +0x9bd
net/http.(*Client).Get(0xbdff20, 0x131eb0a0, 0x63, 0x2, 0x0, 0x0)
        /usr/local/Cellar/go/1.5.1/libexec/src/net/http/client.go:306 +0x92
net/http.Get(0x131eb0a0, 0x63, 0x2, 0x0, 0x0)
        /usr/local/Cellar/go/1.5.1/libexec/src/net/http/client.go:281 +0x3f
github.com/getgauge/gauge/util.Download(0x131eb0a0, 0x63, 0x1324c730, 0x42, 0x13203260, 0x18, 0x1, 0x0, 0x0, 0x0, ...)
        /tmp/src/github.com/getgauge/gauge/util/httpUtils.go:64 +0x30b
github.com/getgauge/gauge/plugin/install.getInstallDescription(0x131356b0, 0xb, 0x1, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0)
        /tmp/src/github.com/getgauge/gauge/plugin/install/install.go:360 +0x31a
github.com/getgauge/gauge/plugin/install.checkPluginUpdates(0x0, 0x0, 0x0)
        /tmp/src/github.com/getgauge/gauge/plugin/install/check.go:125 +0x180
github.com/getgauge/gauge/plugin/install.checkUpdates(0x0, 0x0, 0x0)
        /tmp/src/github.com/getgauge/gauge/plugin/install/check.go:64 +0x48
github.com/getgauge/gauge/plugin/install.printUpdateInfo.func1(0x13134040)
        /tmp/src/github.com/getgauge/gauge/plugin/install/check.go:70 +0x1a
created by github.com/getgauge/gauge/plugin/install.printUpdateInfo
        /tmp/src/github.com/getgauge/gauge/plugin/install/check.go:74 +0x52

goroutine 20 [select]:
github.com/getgauge/gauge/conn.(*GaugeConnectionHandler).acceptConnectionWithoutTimeout(0x13174af0, 0x0, 0x0, 0x0, 0x0)
        /tmp/src/github.com/getgauge/gauge/conn/connectionHandler.go:88 +0x1e9
github.com/getgauge/gauge/conn.(*GaugeConnectionHandler).HandleMultipleConnections(0x13174af0)
        /tmp/src/github.com/getgauge/gauge/conn/connectionHandler.go:136 +0x21
created by github.com/getgauge/gauge/api.startAPIService
        /tmp/src/github.com/getgauge/gauge/api/api.go:58 +0x403

goroutine 29 [IO wait]:
net.runtime_pollWait(0x33261f08, 0x72, 0x131f9d20)
        /usr/local/Cellar/go/1.5.1/libexec/src/runtime/netpoll.go:157 +0x55
net.(*pollDesc).Wait(0x13160b50, 0x72, 0x0, 0x0)
        /usr/local/Cellar/go/1.5.1/libexec/src/net/fd_poll_runtime.go:73 +0x35
net.(*ioSrv).ExecIO(0x13138bd8, 0x13160ab8, 0x94a590, 0x8, 0x131f9d10, 0x13161dc0, 0x0, 0x0)
        /usr/local/Cellar/go/1.5.1/libexec/src/net/fd_windows.go:182 +0x122
net.(*netFD).acceptOne(0x13160a80, 0x13161ce0, 0x2, 0x2, 0x13160ab8, 0x967710, 0x0, 0x0)
        /usr/local/Cellar/go/1.5.1/libexec/src/net/fd_windows.go:564 +0x1d4
net.(*netFD).accept(0x13160a80, 0x0, 0x0, 0x0)
        /usr/local/Cellar/go/1.5.1/libexec/src/net/fd_windows.go:594 +0x11f
net.(*TCPListener).AcceptTCP(0x13138be8, 0x1312e0f0, 0x0, 0x0)
        /usr/local/Cellar/go/1.5.1/libexec/src/net/tcpsock_posix.go:254 +0x47
net.(*TCPListener).Accept(0x13138be8, 0x0, 0x0, 0x0, 0x0)
        /usr/local/Cellar/go/1.5.1/libexec/src/net/tcpsock_posix.go:264 +0x39
github.com/getgauge/gauge/conn.(*GaugeConnectionHandler).acceptConnectionWithoutTimeout.func1(0x13174af0, 0x131ecbc0, 0x131ecc00)
        /tmp/src/github.com/getgauge/gauge/conn/connectionHandler.go:79 +0x27
created by github.com/getgauge/gauge/conn.(*GaugeConnectionHandler).acceptConnectionWithoutTimeout
        /tmp/src/github.com/getgauge/gauge/conn/connectionHandler.go:86 +0xb4

goroutine 21 [syscall, locked to thread]:
syscall.Syscall6(0x767f6bb0, 0x5, 0x258, 0x131d2000, 0x8000, 0x131a5e58, 0x0, 0x0, 0x2a67468, 0x33120030, ...)
        /usr/local/Cellar/go/1.5.1/libexec/src/runtime/syscall_windows.go:139 +0x4a
syscall.ReadFile(0x258, 0x131d2000, 0x8000, 0x8000, 0x131a5e58, 0x0, 0x0, 0x0)
        /usr/local/Cellar/go/1.5.1/libexec/src/syscall/zsyscall_windows.go:283 +0x9d
syscall.Read(0x258, 0x131d2000, 0x8000, 0x8000, 0x33262098, 0x0, 0x0)
        /usr/local/Cellar/go/1.5.1/libexec/src/syscall/syscall_windows.go:299 +0x61
os.(*File).read(0x13138c58, 0x131d2000, 0x8000, 0x8000, 0x0, 0x0, 0x0)
        /usr/local/Cellar/go/1.5.1/libexec/src/os/file_windows.go:300 +0x103
os.(*File).Read(0x13138c58, 0x131d2000, 0x8000, 0x8000, 0x9, 0x0, 0x0)
        /usr/local/Cellar/go/1.5.1/libexec/src/os/file.go:95 +0x6f
io.copyBuffer(0x33262068, 0x1316e5a0, 0x33260ee0, 0x13138c58, 0x131d2000, 0x8000, 0x8000, 0x3c, 0x0, 0x0, ...)
        /usr/local/Cellar/go/1.5.1/libexec/src/io/io.go:381 +0x1da
io.Copy(0x33262068, 0x1316e5a0, 0x33260ee0, 0x13138c58, 0x0, 0x0, 0x0, 0x0)
        /usr/local/Cellar/go/1.5.1/libexec/src/io/io.go:351 +0x53
os/exec.(*Cmd).writerDescriptor.func1(0x0, 0x0)
        /usr/local/Cellar/go/1.5.1/libexec/src/os/exec/exec.go:232 +0x72
os/exec.(*Cmd).Start.func1(0x13178aa0, 0x13175860)
        /usr/local/Cellar/go/1.5.1/libexec/src/os/exec/exec.go:340 +0x1d
created by os/exec.(*Cmd).Start
        /usr/local/Cellar/go/1.5.1/libexec/src/os/exec/exec.go:341 +0x874

goroutine 22 [chan receive]:
github.com/getgauge/gauge/runner.startRunner.func1(0x1313ebc0, 0x13178aa0)
        /tmp/src/github.com/getgauge/gauge/runner/runner.go:181 +0x3e
created by github.com/getgauge/gauge/runner.startRunner
        /tmp/src/github.com/getgauge/gauge/runner/runner.go:184 +0x3f9

goroutine 23 [syscall, locked to thread]:
syscall.Syscall(0x767f6820, 0x2, 0x2cc, 0xffffffff, 0x0, 0x0, 0x0, 0x0)
        /usr/local/Cellar/go/1.5.1/libexec/src/runtime/syscall_windows.go:128 +0x4a
syscall.WaitForSingleObject(0x2cc, 0xffffffff, 0x0, 0x0, 0x0)
        /usr/local/Cellar/go/1.5.1/libexec/src/syscall/zsyscall_windows.go:690 +0x5f
os.(*Process).wait(0x13175900, 0x0, 0x0, 0x0)
        /usr/local/Cellar/go/1.5.1/libexec/src/os/exec_windows.go:18 +0x8b
os.(*Process).Wait(0x13175900, 0x0, 0x0, 0x0)
        /usr/local/Cellar/go/1.5.1/libexec/src/os/doc.go:45 +0x2b
github.com/getgauge/gauge/runner.(*TestRunner).waitAndGetErrorMessage.func1(0x1317bc20)
        /tmp/src/github.com/getgauge/gauge/runner/runner.go:210 +0x2b
created by github.com/getgauge/gauge/runner.(*TestRunner).waitAndGetErrorMessage
        /tmp/src/github.com/getgauge/gauge/runner/runner.go:218 +0x32

goroutine 26 [select]:
net/http.(*persistConn).readLoop(0x1318e240)
        /usr/local/Cellar/go/1.5.1/libexec/src/net/http/transport.go:976 +0x8e8
created by net/http.(*Transport).dialConn
        /usr/local/Cellar/go/1.5.1/libexec/src/net/http/transport.go:685 +0xabc

goroutine 27 [select]:
net/http.(*persistConn).writeLoop(0x1318e240)
        /usr/local/Cellar/go/1.5.1/libexec/src/net/http/transport.go:1009 +0x303
created by net/http.(*Transport).dialConn
        /usr/local/Cellar/go/1.5.1/libexec/src/net/http/transport.go:686 +0xadc

**Comment by mahendrakariya** [comment]
@fthoms Could you please let us know what version of C# plugin you are using? If it's an older version (latest is 0.7.2), could you please run `gauge --update-all` and try again (in a fresh folder)?

**Comment by fthoms** [comment]
@mahendrakariya I downloaded and installed the official installer v. 0.4.0 (64bit and 32 bit) from http://getgauge.io/get-started/index.html. `gauge --version` yields the following results:
`Gauge version: 0.4.0

Plugins
-------
csharp (0.5.1)
html-report (2.1.0)
java (0.3.4)`

A `gauge --update-all` fails with the following:
`Updating plugin 'csharp'
Gathering metadata for csharp
Failed to update plugin 'csharp'.
Reason: Invalid plugin : Could not download csharp-install.json file. Get https://downloads.getgauge.io/plugin/csharp?l=&p=csharp,html-report,java&o=windows&a=amd64: dial tcp 65.49.44.130:443: connectex: A connection attempt failed because the connected party did not properly respond after a period of time, or established connection failed because connected host has failed to respond.

Updating plugin 'html-report'
Gathering metadata for html-report
Failed to update plugin 'html-report'.
Reason: Invalid plugin : Could not download html-report-install.json file. Get https://downloads.getgauge.io/plugin/html-report?l=&p=csharp,html-report,java&o=windows&a=amd64: dial tcp 65.49.44.130:443: connectex: A connection attempt failed because the connected party did not properly respond after a period of time, or established connection failed because connected host has failed to respond.

Updating plugin 'java'
Gathering metadata for java
Failed to update plugin 'java'.
Reason: Invalid plugin : Could not download java-install.json file. Get https://downloads.getgauge.io/plugin/java?l=&p=csharp,html-report,java&o=windows&a=amd64: dial tcp 65.49.44.130:443: connectex: A connection attempt failed because the connected party did not properly respond after a period of time, or established connection failed because connected host has failed to respond.

Failed to update 'csharp, html-report, java' plugins.`

**Comment by fthoms** [comment]
Whoops, it appears that the fault lies partly with me: I have previously used the chocolatey installer (which didn't work, by the way). That, and the installers (.exe) I have tried since then does NOT clean up properly.
I manually deleted c:\program files\gauge and %home%\appdata\local\gauge. After that I could install again, and now it works properly!

Sorry for the inconvenience. The take-away is that the installer should clean up properly when uninstalling ;-)

Keep up the great work!
